### PR TITLE
Port Tests to Core

### PR DESCRIPTION
**Pull request by alinapopa**
Port Castle.Facilities.EventWiring to Core
Port Castle.Facilities.FactorySupport to Core
Exclude Logging and Synchronize tests from building on Core
Port tests

### CODE REVIEW

File: src/Castle.Windsor.Tests/GenericResolutionDeadlockTestCase.cs [hunk 1]
```diff
@@ -106,7 +107,7 @@ public Mapper(IMap<TSource, TDestination> map)
 				this.map = map;
 			}
 
-			#region IMapper<TSource,TDestination> Members
+#region IMapper<TSource,TDestination> Members
```
Comment (jeremymeng) [inline src/Castle.Windsor.Tests/GenericResolutionDeadlockTestCase.cs]: Restore the original indentation

File: src/Castle.Windsor.Tests/GenericResolutionDeadlockTestCase.cs [hunk 2]
```diff
@@ -116,7 +117,7 @@ public void Map(TSource source, TDestination destination)
 				}
 			}
 
-			#endregion
+#endregion
```
Comment (jeremymeng) [inline src/Castle.Windsor.Tests/GenericResolutionDeadlockTestCase.cs]: Restore the original indentation

File: src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+		public static InterfaceMapping GetInterfaceMap(this Type implType, Type interfaceType)
+		{
+			InterfaceMapping map;
+			map.TargetType = implType;
+			map.InterfaceType = interfaceType;
+
+			map.TargetMethods = implType.GetMethods().ToArray();
+			map.InterfaceMethods = new MethodInfo[map.TargetMethods.Length];
+			for(int i=0; i< map.TargetMethods.Length; i++)
```
Comment (jeremymeng) [inline src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs]: whitespaces around oeprators

File: src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+			{
+				map.InterfaceMethods[i] = GetCorrespondingMethodFromType(interfaceType, map.TargetMethods[i]);
+			}
+			return map;
+		}
+
+		// Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+		// that matches in name and parameter types.  Null is returned if there is no match.
+		//From https://github.com/dotnet/wcf/blob/master/src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs
```
Comment (jonorossi) [inline src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs]: Just a reminder that Windsor is Apache 2 licensed, that code is MIT licensed so unless you are contributing it under the Apache 2 license as a Microsoft employee we can't take it.

File: src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+using System.Reflection;
+using System.Text;
+
+#if FEATURE_NETCORE_REFLECTION_API
+namespace System.Reflection
+{
+	internal static class InterfaceMappingReflectionExtensions
+	{
+
+		public static InterfaceMapping GetInterfaceMap(this Type implType, Type interfaceType)
```
Comment (jonorossi) [inline src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs]: Is there a reason we can't use `GetRuntimeInterfaceMap` on .NET Core like we did in Castle Core (https://github.com/castleproject/Core/blob/master/src/Castle.Core/Compatibility/RuntimeReflectionExtensions.cs)?

File: src/Castle.Windsor.Tests/GenericResolutionDeadlockTestCase.cs (hunk 1 above)
Comment (alinapopa) [inline src/Castle.Windsor.Tests/GenericResolutionDeadlockTestCase.cs]: Fixed

File: src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs (hunk 5 above)
Comment (alinapopa) [inline src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs]: Looks like I missed this one. Replacing with `GetRuntimeInterfaceMap`

File: src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs (hunk 4 above)
Comment (alinapopa) [inline src/Castle.Windsor.Tests/Compatibility/InterfaceMapping.cs]: Removed and replaced with `GetInterfaceMap`

**Pull request merged**